File: src/main/java/capricorn/q/hbase/PhoenixUtil.java
package capricorn.q.hbase;

import java.sql.Connection;
import java.sql.DriverManager;
import java.util.LinkedList;

/**
 * @Description : PhoenixUtil
 * @Author : Capricorn.QBB
 * @Date : 2019-04-04
 * @Version : 1.0
 */
public class PhoenixUtil {

	private static LinkedList<Connection> connectionQueue;
	private static final int QUEUE_LEN = 5;

	static {
		try {
			Class.forName("org.apache.phoenix.jdbc.PhoenixDriver");
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		}
	}

	public synchronized static Connection getConnection(String zkUrl) {
		try {
			if (connectionQueue == null) {
				connectionQueue = new LinkedList<>();
				for (int i = 0; i < QUEUE_LEN; i++) {
					Connection conn = DriverManager.getConnection("jdbc:phoenix:" + zkUrl);
					connectionQueue.push(conn);
				}
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		return connectionQueue.poll();
	}

	public static void returnConnection(Connection conn) {
		connectionQueue.push(conn);
	}
}

File: src/main/scala/capricorn/q/hbase/phoenix/PhoenixUtil.scala
package capricorn.q.hbase.phoenix

import java.sql.{Connection, DriverManager}
import java.util

/**
  *
  * @Description : PhoenixUtil
  * @Author : Capricorn.QBB
  * @Date : 2019-04-05
  * @Version : 1.0
  */
object PhoenixUtil {

  private var connectionQueue: util.LinkedList[Connection] = _
  private val QUEUE_LEN = 5

  def getConnection(zkUrl: String): Connection = synchronized {
    try
        if (connectionQueue == null) {
          Class.forName("org.apache.phoenix.jdbc.PhoenixDriver")
          connectionQueue = new util.LinkedList[Connection]
          for (i <- 0 until QUEUE_LEN)
            connectionQueue.push(DriverManager.getConnection("jdbc:phoenix:" + zkUrl))
        }
    catch {
      case e: Exception =>
        e.printStackTrace()
    }
    connectionQueue.poll
  }

  def returnConnection(conn: Connection): Unit = {
    connectionQueue.push(conn)
  }


}
